packages/react-i18n/src/errors.ts

```diff
@@ -1,12 +1,45 @@
-export class MissingTranslationError extends Error {}
-export class MissingReplacementError extends Error {}
-export class MissingCurrencyCodeError extends Error {}
-export class MissingCountryError extends Error {}
-export class InvalidI18nConnectionError extends Error {}
+import {Replacements} from './types';
+
+export class MissingTranslationError extends Error {
+  constructor(key: string) {
+    super(`Missing translation for key: ${key}`);
+  }
+}
+export class MissingReplacementError extends Error {
+  constructor(replacement: string, replacements: Replacements = {}) {
+    const errorMessages = [`No replacement found for key '${replacement}'.`];
+    const replacementKeys = Object.keys(replacements);
+
+    if (replacementKeys.length < 1) {
+      errorMessages.push('And no replacements were passed in.');
+    } else {
+      errorMessages.push(
+        `The following replacements were passed: ${replacementKeys
+          .map(key => `'${key}'`)
+          .join(', ')}`,
+      );
+    }
+
+    super(errorMessages.join(' '));
+  }
+}
+export class MissingCurrencyCodeError extends Error {
+  constructor(scopeName: string) {
+    super(
+      `No currency code provided. ${scopeName} cannot be called without a currency code.`,
```

Comment:
This feels too specific to where we currently throw the error, I’d rather the consumer just pass the whole message.

Reply:
Why would it be too specific? I feel that if the error being thrown is not meant to be specific with a consistent message, we can just throw a generic `Error` with the customized message instead of `MissingCurrencyCodeError`.

Reply:
Because now it is tied to only things that the user actually called. What if the logic for throwing the error goes into a function that doesn't know about the call site? What if we move checking for the timezone to initialization instead of call time?

Reply:
Alternatively, I can enforce only the `No currency code provided.` message, and concat it with whatever the consumer want to display as the second sentence.